pkg/registry/controller/etcd/etcd.go

```diff
@@ -37,7 +38,7 @@ func NewREST(s storage.Interface, storageFactory storage.StorageFactory) (*REST,
 
 	newListFunc := func() runtime.Object { return &api.ReplicationControllerList{} }
 	storageInterface := storageFactory(
-		s, 100, nil, &api.ReplicationController{}, prefix, true, newListFunc)
+		s, 100, etcdstorage.APIObjectVersioner{}, &api.ReplicationController{}, prefix, true, newListFunc)
```

Comment:
Build it into the storage factory function? (and the "no decoration" function, too) I guess that's mostly option 2.